### PR TITLE
[T3A3][T09-A3]Chew Wan Theng Ruth

### PR DESCRIPTION
**Pull request by chewwt**

### CODE REVIEW

**Comment by chewwt** [comment]
ready for review

File: src/seedu/addressbook/data/person/Address.java [hunk 1]
```diff
@@ -25,34 +29,85 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
         if (!isValidAddress(address)) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
-        this.value = address;
+        
+        extractAddressInfo(address);
```
Comment (mauris) [inline src/seedu/addressbook/data/person/Address.java]: good job slapping this part of code into a method on its own.

File: src/seedu/addressbook/data/person/Address.java [hunk 2]
```diff
@@ -25,34 +29,85 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
...
     }
 
     /**
+     * Tries to extract individual information from address
+     *
+     * @throws IllegalValueException if unable to extract from given address
+     */
+	private void extractAddressInfo(String address) throws IllegalValueException {
+		String[] parts = address.split(",");
+        if (parts.length != 4) {
```
Comment (mauris) [inline src/seedu/addressbook/data/person/Address.java]: do take note of indentation

File: src/seedu/addressbook/data/person/Address.java [hunk 3]
```diff
@@ -25,34 +29,85 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
...
+//    	catch (IllegalValueException e) {
+//    	}
+	}
+
+    /**
      * Returns true if a given string is a valid person email.
      */
     public static boolean isValidAddress(String test) {
-        return test.matches(ADDRESS_VALIDATION_REGEX);
+    	 return test.matches(ADDRESS_VALIDATION_REGEX);
```
Comment (mauris) [inline src/seedu/addressbook/data/person/Address.java]: do you think we can run validation on each individual components instead?

File: src/seedu/addressbook/data/person/Address.java [hunk 4]
```diff
@@ -25,34 +29,85 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
...
     }
 
     @Override
     public String toString() {
-        return value;
+    	return this.block + ", " + this.street + ", " + this.unit + ", " + this.postal;
```
Comment (mauris) [inline src/seedu/addressbook/data/person/Address.java]: good work reconstructing and ensuring we don't store duplicates of data

**Comment by mauris** [comment]
Some comments added. Feel free to comment if you have any questions. Once done, you may the PR after reading comments.

**Pull request closed without merging**